### PR TITLE
Add docker/build to github actions

### LINKED ISSUE #2236: Add docker/build to github actions CI so we don't break it

### PR DESCRIPTION
**Pull request by librod89**
### What github issue is this PR for, if any?
Resolves #2236

### What changed, and why?
- Add GitHub action for `docker/build`
- Fixed some unit test failures and lint failures from `main`

### How will this affect user permissions?
N/A

### How is this tested? (please write tests!) 💖💪
N/A

### CODE REVIEW

**Comment by compwron** [comment]
This is looking good! I am EXCITE!

**Comment by librod89** [comment]
@compwron the only thing left to fix is to actually show a failed state for all the failing tests 😆 🤔

**Comment by librod89** [comment]
@compwron I am done working on this. I think the docker test failures should be fixed as part of #2246. What do you think?

File: app/controllers/all_casa_admins_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def edit
   end
 
   def create
-    service = ::CreateAllCasaAdminService.new(params)
+    service = ::CreateAllCasaAdminService.new(params, current_user)
```
Comment (compwron) [inline app/controllers/all_casa_admins_controller.rb:14]: Why does this need doing for docker?

**Review by compwron — APPROVED** [review]

File: app/controllers/all_casa_admins_controller.rb (hunk 1 above)
Comment (librod89) [reply app/controllers/all_casa_admins_controller.rb:14]: All of this was breaking the RSpec build in `main`. I can remove from this PR if you want.

**Comment by compwron** [comment]
I don't love merging a PR that will make part of the build always fail now... but better to have it than not have it! We will fix the tests asap 😅

**Comment by compwron** [comment]
Yay docker in CI!!!!!

**Comment by librod89** [comment]
> I don't love merging a PR that will make part of the build always fail now... but better to have it than not have it! We will fix the tests asap 😅

I agree! But hopefully the tests won't take too long to fix 🤞

**Pull request merged**